internal/impl/ollama/base_processor.go

```diff
@@ -55,22 +65,109 @@ func (co *commandOutput) Write(b []byte) (int, error) {
 
 type key int
 
-const loggerKey key = key(1)
+const (
+	configKey key = key(1)
+)
+
+type runOllamaConfig struct {
+	logger      *service.Logger
+	fs          *service.FS
+	cacheDir    string
+	downloadURL string
+}
+
+func (c *runOllamaConfig) lookPath(file string) (string, error) {
+	if runtime.GOOS != "linux" && runtime.GOOS != "darwin" {
+		return "", errors.ErrUnsupported
+	}
+	path := os.Getenv("PATH")
+	for _, dir := range filepath.SplitList(path) {
+		if !filepath.IsAbs(path) {
+			continue
+		}
+		path := filepath.Join(dir, file)
+		p, err := c.fs.Stat(path)
+		if err != nil || p.IsDir() {
+			continue
+		}
+		// Check that the file is executable
+		if err := syscall.Access(path, 0x1); err == nil {
+			return path, nil
+		}
+	}
+	return "", exec.ErrNotFound
```

Comment:
If possible, can we have an explanation here on why we have to traverse `$PATH` manually and check if the file is executable? I just tested https://pkg.go.dev/os/exec#LookPath and I'm not sure how that function differs.

Reply:
Yeah, the difference is that we're using our filesystem abstractions - will add a comment

Reply:
Might be a good idea to have it in Benthos. Smth to check with @Jeffail for a future refactor